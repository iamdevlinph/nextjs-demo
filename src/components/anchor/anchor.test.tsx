import { shallow } from 'enzyme';
import React from 'react';

import { Anchor } from './index';

describe('Anchor Component', () => {
  it('should match anchor snapshot with only href', () => {
    const wrapper = shallow(<Anchor href="/">lorem</Anchor>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match anchor snapshot if props.block', () => {
    const wrapper = shallow(
      <Anchor href="/" block>
        lorem
      </Anchor>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match anchor snapshot if props.external', () => {
    const wrapper = shallow(
      <Anchor href="/" external>
        lorem
      </Anchor>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match anchor snapshot if props.className', () => {
    const wrapper = shallow(
      <Anchor href="/" className="text-white">
        lorem
      </Anchor>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should match anchor snapshot if children is not a string', () => {
    const wrapper = shallow(
      <Anchor href="/" className="text-white">
        <span>Some text</span>
      </Anchor>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should match anchor snapshot if props.active', () => {
    const wrapper = shallow(
      <Anchor href="/" active>
        lorem
      </Anchor>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
