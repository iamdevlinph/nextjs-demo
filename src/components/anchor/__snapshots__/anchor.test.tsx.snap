// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Component should match anchor snapshot if children is not a string 1`] = `
<Link
  href="/"
>
  <a
    className="text-white"
  >
    <span>
      Some text
    </span>
  </a>
</Link>
`;

exports[`Anchor Component should match anchor snapshot if props.active 1`] = `
<Link
  href="/"
>
  <div
    className="inline-block"
  >
    <a
      className="text-red-900 dark:text-red-300 hover:bg-red-300 hover:text-black cursor-pointer transition ease-in duration-150 underline"
      href="/"
      target=""
    >
      lorem
    </a>
  </div>
</Link>
`;

exports[`Anchor Component should match anchor snapshot if props.block 1`] = `
<Link
  href="/"
>
  <div
    className="block"
  >
    <a
      className="text-red-900 dark:text-red-300 hover:bg-red-300 hover:text-black cursor-pointer transition ease-in duration-150"
      href="/"
      target=""
    >
      lorem
    </a>
  </div>
</Link>
`;

exports[`Anchor Component should match anchor snapshot if props.className 1`] = `
<Link
  href="/"
>
  <div
    className="inline-block"
  >
    <a
      className="text-white cursor-pointer transition ease-in duration-150"
      href="/"
      target=""
    >
      lorem
    </a>
  </div>
</Link>
`;

exports[`Anchor Component should match anchor snapshot if props.external 1`] = `
<div
  className="inline-block"
>
  <a
    className="text-red-900 dark:text-red-300 hover:bg-red-300 hover:text-black cursor-pointer transition ease-in duration-150"
    href="/"
    target="_blank"
  >
    lorem
  </a>
</div>
`;

exports[`Anchor Component should match anchor snapshot with only href 1`] = `
<Link
  href="/"
>
  <div
    className="inline-block"
  >
    <a
      className="text-red-900 dark:text-red-300 hover:bg-red-300 hover:text-black cursor-pointer transition ease-in duration-150"
      href="/"
      target=""
    >
      lorem
    </a>
  </div>
</Link>
`;
